src/wp-includes/cache-compat.php

```diff
@@ -141,3 +141,47 @@ function wp_cache_flush_runtime() {
 		return wp_using_ext_object_cache() ? false : wp_cache_flush();
 	}
 endif;
+
+if ( ! function_exists( 'wp_cache_flush_group' ) ) :
+	/**
+	 * Removes all cache items in a group, if the object cache implementation supports it.
+	 * Before calling this method, always check for group flushing support using the
+	 * `wp_cache_supports_group_flush()` method.
+	 *
+	 * @since 6.1.0
+	 *
+	 * @see WP_Object_Cache::flush_group()
+	 * @global WP_Object_Cache $wp_object_cache Object cache global instance.
+	 *
+	 * @param string|array $group name(s) of group to remove from cache.
+	 * @return bool|array|WP_Error Bool or array of bool if array passed, WP_Error if not supported.
+	 */
+	function wp_cache_flush_group( $group ) {
+		global $wp_object_cache;
+
+		if ( ! wp_cache_supports_group_flush() ) {
+			$error = new WP_Error( 'unsupported', __( 'Your object cache implementation does not support flushing individual groups.' ) );
+
+			_doing_it_wrong( __FUNCTION__, $error->get_error_message(), '6.1.0' );
+
+			return $error;
```

Comment:
Let's not return an error here. `_doing_it_wrong()` sounds good, but no other cache functions return `WP_Error`, so I think we should stick with the simple `bool` for consistency.
```suggestion
			_doing_it_wrong( __FUNCTION__, __( 'Your object cache implementation does not support flushing individual groups.' ), '6.1.0' );

			return false;
```

Reply:
This was suggested in #2368, to safeguard against users not checking `wp_cache_supports_group_flush()` before using the feature.

I'd suggest leaving it to avoid users calling the method expecting it to work, because few will check the return type.

Reply:
@spacedmonkey Any feedback?

Reply:
I think the `_doing_it_wrong()` should be obvious enough - that's way more visible than either a `bool` or a `WP_Error` return type.

My main concern with a `WP_Error` here is that developers may be used to check for `bool` return values on the cache functions for success/failure, and for those suddenly having a `WP_Error` here would be problematic as it would evaluate to `true`, and it would be inconsistent (and inconvenient) to always have to check `if ( $result && ! is_wp_error( $result ) )`.

In any case, I think we shouldn't optimize our code for the people that are `_doing_it_wrong`, but rather for those that do it right :)

Reply:
Ah, I see, that makes sense.

Reply:
Resolved in https://github.com/WordPress/wordpress-develop/pull/2969/commits/c53957b9a06b503ed3e777cca3f53d561981a900